e2e/common/common.go

```diff
@@ -35,28 +46,90 @@ func NewConfig(
 	validators, account, numEthAccounts int,
 	useFnConsensus bool,
 ) (*lib.Config, error) {
-	basedirAbs, err := filepath.Abs(path.Join(BaseDir, name))
+	checkAppHashAfterEV := os.Getenv(checkAppHashAfterCmds)
+	checkAppHashAfterCmds := len(checkAppHashAfterEV) > 0
+
+	LoomPath := os.Getenv(loomExeEv)
+	if len(LoomPath) == 0 {
+		LoomPath = DefaultLoomPath
+	}
+
+	LoomPath2 := os.Getenv(loomExe2Ev)
+	if len(LoomPath2) == 0 {
+		LoomPath2 = DefatulLoomPath2
+	}
+
+	var err error
+	evValidators := uint64(0)
+	validatorsEv := os.Getenv(ValidatorsEv)
+	if len(validatorsEv) > 0 {
+		evValidators, err = strconv.ParseUint(validatorsEv, 0, 64)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	evAltValidators := uint64(0)
+	altValidatorsEv := os.Getenv(AltValidatorsEv)
+	if len(altValidatorsEv) > 0 {
+		evAltValidators, err = strconv.ParseUint(altValidatorsEv, 0, 32)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	v, altV, err := splitValidators(uint64(validators), evValidators, evAltValidators)
+
+	return GenerateConfig(
+		name, testFile, genesisTmpl, yamlFile, BaseDir, ContractDir, LoomPath, LoomPath2,
+		int(v), int(altV), account, numEthAccounts,
+		useFnConsensus, *Force, checkAppHashAfterCmds,
+	)
+}
+
+func splitValidators(validators, evValidators, evAltValidators uint64) (uint64, uint64, error) {
+	if validators == 0 {
+		return 0, 0, nil
+	}
+
+	if validators == 1 {
+		return 1, 0, nil
+	}
+
+	if validators >= evValidators+evAltValidators {
+		return validators - evAltValidators, evAltValidators, nil
+	}
+
+	if validators <= evValidators+1 {
```

Comment:
So with this logic if validators = 3, evValidator = 1, evAltValidator = 3, we will get 1, 1 which overrides the number of validators

Reply:
if validators = 3, evValidator = 1, evAltValidator = 3 this check fails as !(3<= 1+1), but if it did not fail the return would be (3-1, 1) = (2,1). The actual values returned is from the last line`evValidators, validators - evValidators` ie (1, 3-1) = (1,2).
Adding the returned values should always produce validators e.g. `validators - 1 + 1 = validators`